event/engine.go

```diff
@@ -0,0 +1,17 @@
+package event
+
+// EngineEventType type to describe engine events
+// Engine events doesn't need the same validation that normal events need
+// These events are used internally only and cannot be emitted by services
+type EngineEventType string
+
+const (
+	// EngineAPIExecution This event is triggered when the `execution/create` API is called
+	EngineAPIExecution EngineEventType = "mesg:engine-api-execution"
+)
+
+// EngineEvent creates an engine event.
+// Engine event does not have any instance hash and are prefixed by mesg:
+func EngineEvent(evtType EngineEventType, data map[string]interface{}) *Event {
```

Comment:
This function should be prefixed with `New`, shouldn't it?
```suggestion
func New EngineEvent(evtType EngineEventType, data map[string]interface{}) *Event {
```